src/lib/relicScorer.ts

```diff
@@ -597,98 +654,123 @@ function getMainStatWeight(relic, scoringMetadata) {
 }
 
 // Create a fake relic to feed into score() to get accurate scores for potential relics
-function fakeRelic(grade: RelicGrade, enhance: RelicEnhance, part: string, mainstat: string, substats){
-  let fake : Relic = {
+function fakeRelic(grade: RelicGrade, enhance: RelicEnhance, part: string, mainstat: string, substats) {
+  const fake: Relic = {
     weights: undefined,
     cs: undefined,
     ds: undefined,
     ss: undefined,
-    enhance : 0,
-    equippedBy : "",
-    grade : grade,
-    id: "",
-    set: "",
+    enhance: enhance,
+    equippedBy: '',
+    grade: grade,
+    id: '',
+    set: '',
     part: part,
     main: {
       stat: mainstat,
-      value: MainStatsValues[mainstat][grade]["base"]+MainStatsValues[mainstat][grade]["increment"]*enhance
+      value: MainStatsValues[mainstat][grade].base + MainStatsValues[mainstat][grade].increment * enhance,
     },
-    substats: substats
+    substats: substats,
   }
   return fake
 }
 
-//create the substats array to feed into fakeRelic(), addedRolls is left at 0 as it is not used
-function generateSubStats(grade: RelicGrade, sub1: SubStats, val1: number, sub2: SubStats, val2: number, sub3: SubStats, val3: number, sub4: SubStats, val4: number){
-  let substats = [
+// Create the substats array to feed into fakeRelic(), addedRolls is left at 0 as it is not used
+function generateSubStats(grade: RelicGrade, sub1: SubStats, val1: number, sub2: SubStats, val2: number, sub3: SubStats, val3: number, sub4: SubStats, val4: number) {
```

Comment:
Not a huge deal, I see why you did it this way but it might be cleaner to pass in a `substats: Stat[]` instead of the sub/vals, so we dont have to construct the function args like this which is really hard to read/check for errors

```
    const subs = generateSubStats(5, substats[0][0], SubStatValues[substats[0][0]][5].high * 6, substats[1][0], SubStatValues[substats[1][0]][5].high
      , substats[2][0], SubStatValues[substats[2][0]][5].high, substats[3][0], SubStatValues[substats[3][0]][5].high)
```